containerized.mk

```diff
@@ -0,0 +1,46 @@
+IMAGE_NAME=docker/swarmkit
+GOPATH=/go
+DOCKER_IMAGE_DIR=${GOPATH}/src/${PROJECT_ROOT}
+
+# don't bother writing every single make target. just pass the call through to
+# docker and make
+# we prefer `%:` to `.DEFAULT` as the latter doesn't run phony deps
+# (see https://www.gnu.org/software/make/manual/html_node/Special-Targets.html)
+%::
+	@ echo "Running target $@ inside a container"
+	@ DOCKER_SWARMKIT_DOCKER_RUN_CMD="make $*" $(MAKE) run
+
+shell:
+	@ DOCKER_SWARMKIT_DOCKER_RUN_CMD='bash' DOCKER_SWARMKIT_DOCKER_RUN_FLAGS='-i' $(MAKE) run
+
+.PHONY: image
+image:
+	docker build -t ${IMAGE_NAME} .
+
+# internal target, only builds the image if it doesn't exist
+.PHONY: ensure_image_exists
+ensure_image_exists:
+	@ if [ ! $$(docker images -q ${IMAGE_NAME}) ]; then $(MAKE) image; fi
+
+# internal target, starts the sync if needed
+# uses https://github.com/EugenMayer/docker-sync/blob/47363ee31b71810a60b05822b9c4bd2176951ce8/tasks/sync/sync.thor#L193-L196
+# which is not great, but that's all they expose so far to do this...
+.PHONY: ensure_sync_started
+ensure_sync_started:
+	@ kill -0 $$(cat .docker-sync/daemon.pid) 2&> /dev/null || docker-sync start
```

Comment:
can you add a comment explaining in a nutshell what this does? something along the lines of:

```
# checks if the daemon pid in the .docker-sync directory maps
# to the present daemon. if so, then...
```

...and so-on.

Reply:
👍